pkg/kubectl/registry.go

```diff
@@ -11,35 +11,417 @@ type KubectlCommand struct {
+	for _, reg := range toolRegistry {
+		if shouldRegisterTool(reg.minAccess, accessLevel) {
+			tool := createToolFromRegistration(reg, accessLevel)
+			tools = append(tools, tool)
+		}
+	}
+
+	return tools
+}
+
+// isValidAccessLevel checks if the given access level is valid
+func isValidAccessLevel(accessLevel string) bool {
+	return accessLevel == AccessLevelReadOnly ||
+		accessLevel == AccessLevelReadWrite ||
+		accessLevel == AccessLevelAdmin
+}
+
+// shouldRegisterTool determines if a tool should be registered based on access levels
+func shouldRegisterTool(minAccess, currentAccess string) bool {
+	accessLevels := map[string]int{
+		AccessLevelReadOnly:  1,
+		AccessLevelReadWrite: 2,
+		AccessLevelAdmin:     3,
+	}
+
+	minLevel := accessLevels[minAccess]
+	currentLevel := accessLevels[currentAccess]
+
+	return currentLevel >= minLevel
+}
+
+// createToolFromRegistration creates a tool from its registration definition
+func createToolFromRegistration(reg toolRegistration, accessLevel string) mcp.Tool {
+	switch creator := reg.creator.(type) {
+	case toolCreator:
+		// Tools that support read-only mode
+		readOnly := accessLevel == AccessLevelReadOnly && reg.readOnlyMode
+		return creator(readOnly)
+	case toolCreatorSimple:
+		// Tools that don't have read-only variants
+		return creator()
+	default:
+		panic("invalid tool creator type")
+	}
+}
+
+// createResourcesTool creates the main resource management tool
+func createResourcesTool(readOnly bool) mcp.Tool {
+	var description string
+	var operationDesc string
+
+	if readOnly {
+		description = `View Kubernetes resources with read-only operations.
+
+Available operations:
+- get: Display one or many resources
+- describe: Show detailed information about resources
+
+Common resources: pods, deployments, services, configmaps, secrets, namespaces, etc.
+
+Examples:
+- Get pods: operation='get', resource='pods', args='-n default'
+- Describe deployment: operation='describe', resource='deployment', args='myapp -n production'`
+		operationDesc = "The operation to perform: get, describe"
+	} else {
+		description = `Manage Kubernetes resources with standard CRUD operations.
+
+Available operations:
+- get: Display one or many resources
+- describe: Show detailed information about resources
+- create: Create a resource from a file or stdin
+- delete: Delete resources
+- apply: Apply a configuration to a resource
+- patch: Update fields of a resource
+- replace: Replace a resource
+
+Common resources: pods, deployments, services, configmaps, secrets, namespaces, etc.
+
+Examples:
+- Get pods: operation='get', resource='pods', args='-n default'
+- Describe deployment: operation='describe', resource='deployment', args='myapp -n production'
+- Apply config: operation='apply', resource='', args='-f deployment.yaml'
+- Delete service: operation='delete', resource='service', args='myservice -n default'`
+		operationDesc = "The operation to perform: get, describe, create, delete, apply, patch, replace"
```

Comment:
what is this operationDesc ?

Reply:
It is used here:

<img width="1200" height="423" alt="image" src="https://github.com/user-attachments/assets/2c7dbdbf-7bcb-41a0-9bd2-76e5a3e5dd8b" />